client/src/main/java/io/getstream/chat/android/client/utils/Result.kt

```diff
@@ -2,27 +2,25 @@ package io.getstream.chat.android.client.utils
 
 import io.getstream.chat.android.client.errors.ChatError
 
-data class Result<T>(
+public data class Result<T : Any>(
```

Comment:
Since this class is used as a read-only class by clients, and they never need to instantiate it, should we just hide it behind an interface?